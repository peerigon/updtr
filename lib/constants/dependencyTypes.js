"use strict";

const REGULAR = 0;
const DEV = 1;
const OPTIONAL = 2;

function fromNpm(type) {
    switch (type) {
        case "dependencies":
            return REGULAR;
        case "devDependencies":
            return DEV;
        case "optionalDependencies":
            return OPTIONAL;
        default:
            throw new Error(`Unsupported dependency type: ${ type }`);
    }
}

exports.REGULAR = REGULAR;
exports.DEV = DEV;
exports.OPTIONAL = OPTIONAL;

exports.fromNpm = fromNpm;
// yarn uses the same strings. Change this if necessary.
exports.fromYarn = fromNpm;
