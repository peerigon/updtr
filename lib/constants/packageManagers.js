"use strict";

const NPM = "npm";
const YARN = "yarn";
const SUPPORTED = [NPM, YARN];

exports.NPM = NPM;
exports.YARN = YARN;
exports.SUPPORTED = SUPPORTED;
