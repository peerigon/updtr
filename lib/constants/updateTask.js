"use strict";

const WANTED = "wanted";
const LATEST = "latest";

exports.WANTED = WANTED;
exports.LATEST = LATEST;
