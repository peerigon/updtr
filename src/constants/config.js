// The first value in arrays is the default value

export const USE_NPM = "npm";
export const USE_YARN = "yarn";
export const USE_OPTIONS = [USE_NPM, USE_YARN];
export const UPDATE_TO_LATEST = "latest";
export const UPDATE_TO_NON_BREAKING = "non-breaking";
export const UPDATE_TO_WANTED = "wanted";
export const UPDATE_TO_OPTIONS = [
    UPDATE_TO_LATEST,
    UPDATE_TO_NON_BREAKING,
    UPDATE_TO_WANTED,
];
export const SAVE_SMART = "smart";
export const SAVE_CARET = "caret";
export const SAVE_EXACT = "exact";
export const SAVE_OPTIONS = [SAVE_SMART, SAVE_CARET, SAVE_EXACT];
