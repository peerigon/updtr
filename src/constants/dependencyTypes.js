export const REGULAR = "dependencies";
export const DEV = "devDependencies";
export const OPTIONAL = "optionalDependencies";

export function fromNpm(type) {
    switch (type) {
        case "dependencies":
            return REGULAR;
        case "devDependencies":
            return DEV;
        case "optionalDependencies":
            return OPTIONAL;
        default:
            throw new Error(`Unsupported dependency type: ${ type }`);
    }
}

export {
    // yarn uses the same strings. Change this if necessary.
    fromNpm as fromYarn,
};
