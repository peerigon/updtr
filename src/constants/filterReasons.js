export const GIT = "git";
export const EXOTIC = "exotic";
export const UNSTABLE = "unstable";
export const EXCLUDED = "excluded";
export const NOT_WANTED = "not-wanted";
