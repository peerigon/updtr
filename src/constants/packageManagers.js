export const NPM = "npm";
export const YARN = "yarn";
export const SUPPORTED = [NPM, YARN];
