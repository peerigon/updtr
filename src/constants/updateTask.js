export const UPDATE_TO_WANTED = "wanted";
export const UPDATE_TO_LATEST = "latest";
