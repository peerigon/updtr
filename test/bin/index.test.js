import { execFile } from "child_process";
import path from "path";
import { USE_YARN } from "../../src/constants/config";

const projectPath = path.resolve(__dirname, "..", "..").replace(/\\/g, "/");
const pathToBabelNode = require.resolve("babel-cli/bin/babel-node");
const pathToRunBin = require.resolve("../helpers/runBinMock");

// These tests may take longer on travis
jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;

async function execRunBinMock(
    { cwd = __dirname, args = [], runMock = "" } = {}
) {
    const stdout = await new Promise((resolve, reject) => {
        execFile(
            "node",
            [pathToBabelNode, pathToRunBin, ...args],
            {
                cwd,
                env: {
                    ...process.env,
                    RUN_MOCK: runMock,
                },
            },
            (err, stdout, stderr) =>
                void (err ? reject(err) : resolve(stdout || stderr))
        );
    });

    const stdoutWithoutBasePath = stdout
        // Replace double-backslashes with one forward slash
        .replace(/\\\\/g, "/")
        // Remove project path
        .split(projectPath)
        .join("");

    return JSON.parse(stdoutWithoutBasePath);
}

describe("bin", () => {
    describe("when no arguments have been specified", () => {
        it("should run updtr with the default config", async () => {
            const configs = await execRunBinMock();

            expect(configs).toMatchSnapshot();
        });
    });
    describe("when the binary is executed in a directory with a yarn.lock file", () => {
        it("should use yarn", async () => {
            const cwd = path.resolve(__dirname, "..", "fixtures", "empty");
            const configs = await execRunBinMock({ cwd });

            expect(configs.updtrConfig.use).toBe(USE_YARN);
        });
    });
    describe("when all arguments have been specified", () => {
        it("should run updtr with the expected config", async () => {
            const args = [
                "--reporter",
                "dense",
                "--use",
                "yarn",
                "--exclude",
                "a",
                "b",
                "c",
                "--test",
                "jest -b",
                // yarn does not support custom registries
                // We don't test for this option here and assume that it'll work if all the other options did also work
                // "--registry http://example.com",
                "--update-to",
                "non-breaking",
                "--test-stdout",
                "--save",
                "exact",
            ];
            const configs = await execRunBinMock({ args });

            expect(configs).toMatchSnapshot();
        });
    });
    describe("when there is an error", () => {
        it("should emit an error event on the updtr instance", async () => {
            const error = await execRunBinMock({
                runMock: "rejectWithAccessError",
                args: ["--reporter", "error"],
            });

            expect(error).toMatchSnapshot();
        });
    });
});
