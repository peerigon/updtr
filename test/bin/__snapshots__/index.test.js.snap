// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bin when all arguments have been specified should run updtr with the expected config 1`] = `
Object {
  "reporter": "dense",
  "reporterConfig": Object {
    "stream": null,
    "testStdout": true,
  },
  "updtrConfig": Object {
    "cwd": "/test/bin",
    "exclude": Array [
      "a",
      "b",
      "c",
    ],
    "save": "exact",
    "test": "jest -b",
    "updateTo": "non-breaking",
    "use": "yarn",
  },
}
`;

exports[`bin when no arguments have been specified should run updtr with the default config 1`] = `
Object {
  "reporter": "dense",
  "reporterConfig": Object {
    "stream": null,
    "testStdout": false,
  },
  "updtrConfig": Object {
    "cwd": "/test/bin",
    "exclude": Array [],
    "save": "smart",
    "updateTo": "latest",
    "use": "npm",
  },
}
`;

exports[`bin when there is an error should emit an error event on the updtr instance 1`] = `"Cannot access package.json in /test/bin"`;
